src/cmd/linuxkit/run_vmware.go

```diff
@@ -84,49 +91,74 @@ func runVMware(args []string) {
 	}
 	prefix := remArgs[0]
 
-	// Build the contents of the VMWare .vmx file
-	vmx := buildVMX(*runCPUs, *runMem, *runDisk, prefix)
-
-	if vmx == "" {
-		log.Fatalf("VMware .vmx file could not be generated, please confirm inputs")
-	}
-
-	var path string
+	var vmrunPath, vmDiskManagerPath string
 	var vmrunArgs []string
 
 	if runtime.GOOS == "windows" {
-		path = "C:\\Program\\ files\\VMware Workstation\\vmrun.exe"
+		vmrunPath = "C:\\Program\\ files\\VMware Workstation\\vmrun.exe"
+		vmDiskManagerPath = "C:\\Program\\ files\\VMware Workstation\\vmware-vdiskmanager.exe"
 		vmrunArgs = []string{"-T", "ws", "start", prefix + ".vmx"}
 	}
 
 	if runtime.GOOS == "darwin" {
-		path = "/Applications/VMware Fusion.app/Contents/Library/vmrun"
+		vmrunPath = "/Applications/VMware Fusion.app/Contents/Library/vmrun"
+		vmDiskManagerPath = "/Applications/VMware Fusion.app/Contents/Library/vmware-vdiskmanager"
 		vmrunArgs = []string{"-T", "fusion", "start", prefix + ".vmx"}
 	}
 
 	if runtime.GOOS == "linux" {
-		path = "vmrun"
-		fullVMrunPath, err := exec.LookPath(path)
+		vmrunPath = "vmrun"
+		vmDiskManagerPath = "vmware-vdiskmanager"
+		fullVMrunPath, err := exec.LookPath(vmrunPath)
 		if err != nil {
-			log.Fatalf("Unable to find %s within the $PATH", path)
+			// Kept as seperate error as people may manually change their environment vars
+			log.Fatalf("Unable to find %s within the $PATH", vmrunPath)
 		}
-		path = fullVMrunPath
+		vmrunPath = fullVMrunPath
 		vmrunArgs = []string{"-T", "ws", "start", prefix + ".vmx"}
 	}
 
-	// Check executables exist before attempting to execute
-	if _, err := os.Stat(path); os.IsNotExist(err) {
+	// Check vmrunPath exist before attempting to execute
+	if _, err := os.Stat(vmrunPath); os.IsNotExist(err) {
 		log.Fatalf("ERROR VMware exectuables can not be found, ensure software is installed")
 	}
 
+	if *vmwareDisk != "" {
+		// Check vmDiskManagerPath exist before attempting to execute
+		if _, err := os.Stat(vmDiskManagerPath); os.IsNotExist(err) {
+			log.Fatalf("ERROR VMware Disk Manager exectuables can not be found, ensure software is installed")
+		}
+		*vmwareDisk = *vmwareDisk + ".vmdk"
+
+		// If disk doesn't exist then create one
+		if _, err := os.Stat(*vmwareDisk); err != nil {
+			if os.IsNotExist(err) {
+				log.Infof("Creating new VMware disk [%s]", *vmwareDisk)
+				vmDiskCmd := exec.Command(vmDiskManagerPath, "-c", "-s", *vmwareDiskSize, "-a", "lsilogic", "-t", "0", *vmwareDisk)
+				if err := vmDiskCmd.Run(); err != nil {
+					log.Errorf("Error creating disk [%s]:  %s", *vmwareDisk, err.Error())
+				}
+			}
```

Comment:
same here: should we have an `else` case that bails if the err is not nil and `os.IsNotExist(err) != true`?